### ISSUE TITLE
feature request: Multiple identities per account

### ISSUE THREAD

**Issue by Beethoven-n** [revoltchat/backend#284]
### What do you want to see?

pointer to the start of this discussion: https://github.com/orgs/revoltchat/discussions/76
here's the original text:

> This is a pretty specific request, so bear with me.
> # Summary
> 
> On Discord, there's an ecosystem of bots that exist purely to resend messages using a different username or avatar. Because Discord has no native support for this, they use webhooks - so they appear as follows:
> 
> ![image](https://user-images.githubusercontent.com/204153/131220492-11a542ca-8263-43f7-9ee8-dc62d195f0d8.png)
> 
> This post proposes adding a system for multiple identities per account to Revolt, while the systems are still new and malleable.
> ## Rationale
> 
> There are two major groups I have in mind with this feature.
> ### 1. Plural systems
> 
> For those that aren't aware, there exists a subset of the population that are known as **plural systems**. This term is used to describe people with pluralities - multiple people within one body, so to speak. If you're not familiar with the concept, feel free to take a look at [this primer](https://morethanone.info/). Members of plural systems are sometimes called "alters", if you're already familiar with that term.
> 
> Plurality tools like the bots I've mentioned above are a key point of comfort and communication for members of plural systems. They allow each individual member to interact with an avatar and name of their own, which means they're not constrained to a single identity. However, Discord doesn't go far enough, here - there are strict rate limits on nickname and avatar changes (which would already be incredibly burdensome, especially given how fast switches can happen and the fact that there may be multiple members chatting at once), and no built-in identity system.
> 
> A properly integrated solution is something Revolt could break ground on here, and it would be a massive bonus for all plural systems.
> ### 2. Roleplayers
> 
> Another common use for bots like those I've mentioned above is for role-playing servers. As you can probably tell, having multiple identities available at the push of a button would be extremely convenient.
> ### Niche use-cases
> 
> I could also envision other use-cases here, such as staff members using separate identities for personal and official communications - I'm sure there are other possibilities.
> ## Concept
> 
> There are many ways this could be exposed in the frontend. I'll leave the configuration side of this to the community and developers, but I've had a couple of ideas on how the UX might look:
> 
> ![multi-identities](https://user-images.githubusercontent.com/204153/131222035-b80a3d3b-1dd7-4daa-a724-4f4a3b8b6eae.png)
> 
> You can probably tell I'm not a graphic designer, haha

**Comment by erkinalp** [comment]
This should preferrably be designed to allow M:N mappings from senders to authors, to allow things such as posting in behalf of a group/organisation.

**Comment by icefisher225** [comment]
I would also like to see this as a proper feature and not built via webhook abuse. It would make Revolt my (and many others) default chat app.

**Comment by insertish** [comment]
This is already implemented in the API but lacks any client support currently.
https://github.com/revoltchat/frontend/issues/244

**Comment by erkinalp** [comment]
@insertish No, masquerades are 1:N, we need a true M:N sender→author mapping capability.

**Comment by PilotMiaoumi** [comment]
EDIT: I edited this comment because I wanted to better explain stuff.

I think this is worth reopening. Even though masquerading exists in the API, there’s no real client support, so in practice it’s not usable without workarounds which can be hard to set up.

Implementing this natively would be a big improvement over bots handling it through webhooks and would be easier to use. I'm sure many people will appreciate this, I would like it myself as I would feel recognised. This being a proper part of the platform is just nicer.

I know many who would benefit from this, myself included, and it's the kind of feature that could pull users over to Stoat. There is a demand for it and people are already relying on other tools to use this feature.

If this is to be revisited though, I do feel it is important to support M:N sending, as opposed to 1:N, as it would better reflect how people are already using these tools in practice, and so identities actually feel like distinct people instead of a workaround that only exists per message.

The only way I know how to do this type of thing is Avia, but it is website only (I believe) and in my opinion it is quite hard to set it up, which makes it feel less accessible than this should be. I am especially tech aligned and I had trouble figuring out what exactly to do. My big gripe with this solution as well is that in my view, this is not an implementation and there should be a push to make this official.

basically: I feel this issue deserves more discussion and perhaps reopening.
